genie-web/src/main/java/com/netflix/genie/web/configs/grpc/GenieGRpcServerAutoConfiguration.java

```diff
@@ -0,0 +1,115 @@
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.configs.grpc;
+
+import com.netflix.genie.web.properties.GRpcServerProperties;
+import com.netflix.genie.web.rpc.grpc.servers.GRpcServerManager;
+import io.grpc.BindableService;
+import io.grpc.Server;
+import io.grpc.ServerInterceptor;
+import io.grpc.ServerInterceptors;
+import io.grpc.netty.NettyServerBuilder;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.util.SocketUtils;
+
+import javax.annotation.PostConstruct;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Controls whether a gRPC server is configured and started for this Genie node or not.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Configuration
+@ConditionalOnProperty(value = GRpcServerProperties.ENABLED_PROPERTY, havingValue = "true")
+@EnableConfigurationProperties(
+    {
+        GRpcServerProperties.class,
+    }
+)
+@AutoConfigureAfter(
+    name = {
+        "com.netflix.springboot.grpc.server.GrpcServerAutoConfiguration"
+    }
+)
+@Slf4j
+public class GenieGRpcServerAutoConfiguration {
+
+    private static final int RANDOM_PORT = 0;
+
+    /**
+     * Log that gRPC server is enabled.
+     */
+    @PostConstruct
+    public void postConstruct() {
+        log.info("gRPC server configuration is ENABLED");
+    }
+
+    /**
+     * Create a {@link Server} if one isn't already present in the context.
+     *
+     * @param port               The port this server should listen on
+     * @param services           The gRPC services this server should serve
+     * @param serverInterceptors The {@link ServerInterceptor} implementations that should be applied to all services
+     * @return A Netty server instance based on the provided information
+     */
+    @Bean
+    @ConditionalOnMissingBean(Server.class)
+    public Server gRpcServer(
+        @Value("${grpc.server.port:0}") final int port,
+        final Set<BindableService> services,
+        final List<ServerInterceptor> serverInterceptors
+    ) {
+        final NettyServerBuilder builder = NettyServerBuilder.forPort(
+            port == RANDOM_PORT
+                ? SocketUtils.findAvailableTcpPort()
```

Comment:
I'd prefer to set the port to 0 and let the socket bind to an ephemeral port.  There's no chance of collision that way.

Reply:
Good point I'll fix that